api/src/main/scala/pl/touk/nussknacker/engine/api/process/RunMode.scala

```diff
@@ -3,16 +3,21 @@ package pl.touk.nussknacker.engine.api.process
 /**
   * Specifies the mode a node is invoked. It can be one of the following values:
   * <ul>
-  * <li>Normal - process is executed inside Flink or other engine. Real implementation of node should be used. Note, this
-  * mode is also used for service query, validations and test data generation.</li>
+  * <li>Normal - used when process is executed inside Flink or other engine, and also for service query, validations and test data generation.
+  * In this mode real implementation of node should be used.</li>
   * <li>Test - process is run in test mode to collect results for test data. Can be used to stub real implementation.</li>
   * </ul>
   */
 sealed trait RunMode
 
 object RunMode {
 
-  case object Normal extends RunMode
+  sealed trait Normal extends RunMode
+
+  case object Engine extends Normal
+  case object Validation extends Normal
+  case object ServiceQuery extends Normal
+  case object TestDataGeneration extends Normal
```

Comment:
The same as above . I think that exposing too many modes will cause that consumer of those modes will be confused how they should be handled. Also when we change implementation details, some of those modes will go away and we will be hard to migrate to new approach. Maybe it should be extracted only one new mode ServiceQuery which indeed is in context of running component and validation / testdatageneration should be hidden as a normal mode as a implementation detail? (also removed from documentation)